src/ui.rs

```diff
@@ -2655,24 +2663,96 @@ impl Style {
-                "bold" => s.weight = Some(ftd::Weight::Bold),
-                "semi-bold" => s.weight = Some(ftd::Weight::SemiBold),
-                "medium" => s.weight = Some(ftd::Weight::Medium),
-                "regular" => s.weight = Some(ftd::Weight::Regular),
-                "light" => s.weight = Some(ftd::Weight::Light),
-                "extra-light" => s.weight = Some(ftd::Weight::ExtraLight),
-                "hairline" => s.weight = Some(ftd::Weight::HairLine),
+                "italic" => {
+                    s.italic = true;
+                    add_in_map("italic", &mut booleans);
+                }
+                "underline" => {
+                    s.underline = true;
+                    add_in_map("underline", &mut booleans);
+                }
+                "strike" => {
+                    s.strike = true;
+                    add_in_map("strike", &mut booleans);
+                }
+                "heavy" => {
+                    s.weight = Some(ftd::Weight::Heavy);
+                    add_in_map("heavy", &mut weights);
+                }
+                "extra-bold" => {
+                    s.weight = Some(ftd::Weight::ExtraBold);
+                    add_in_map("extra-bold", &mut weights);
+                }
+                "bold" => {
+                    s.weight = Some(ftd::Weight::Bold);
+                    add_in_map("bold", &mut weights);
+                }
+                "semi-bold" => {
+                    s.weight = Some(ftd::Weight::SemiBold);
+                    add_in_map("semi-bold", &mut weights);
+                }
+                "medium" => {
+                    s.weight = Some(ftd::Weight::Medium);
+                    add_in_map("medium", &mut weights);
+                }
+                "regular" => {
+                    s.weight = Some(ftd::Weight::Regular);
+                    add_in_map("regular", &mut weights);
+                }
+                "light" => {
+                    s.weight = Some(ftd::Weight::Light);
+                    add_in_map("light", &mut weights);
+                }
+                "extra-light" => {
+                    s.weight = Some(ftd::Weight::ExtraLight);
+                    add_in_map("extra-light", &mut weights);
+                }
+                "hairline" => {
+                    s.weight = Some(ftd::Weight::HairLine);
+                    add_in_map("hairline", &mut weights);
+                }
                 t => return ftd::e2(format!("{} is not a valid style", t), doc_id, 0),
             }
         }
+
+        // Checks if there is repeatation in Underline,italic,strike
+        for (style, count) in booleans.iter() {
+            if count > &1 {
+                return Err(ftd::p1::Error::ForbiddenUsage {
+                    message: format!("\'{}\' repeated {} times in \'{}\'", style, count, &l),
+                    doc_id: doc_id.to_string(),
+                    line_number: 0,
```

Comment:
The line number is wrong. Pass line number from the caller (of the from method) if need be.